internal/service/s3/object.go

```diff
@@ -352,18 +358,21 @@ func resourceObjectUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 
 func resourceObjectDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
-	conn := meta.(*conns.AWSClient).S3Conn(ctx)
+	conn := meta.(*conns.AWSClient).S3Client(ctx)
 
 	bucket := d.Get("bucket").(string)
 	key := d.Get("key").(string)
+	// ****************
+	// TODO: REVIEW
```

Comment:
Leftover comment?

Reply:
Yes, this PR is still _WIP_ (updated title).
Still need to address differences with [automatic URI cleaning](https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#hdr-Automatic_URI_cleaning).